documentation/tests/integration/components/list-item-test.js

```diff
@@ -0,0 +1,151 @@
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('div#test-list-item').hasClass('my-class');
+    assert.dom('div#test-list-item').hasAttribute('data-test1');
+    assert.dom('div#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <a> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @href="#" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('a#test-list-item').hasClass('my-class');
+    assert.dom('a#test-list-item').hasAttribute('data-test1');
+    assert.dom('a#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <LinkTo> element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem @route="index" id="test-list-item" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('a#test-list-item').hasClass('my-class');
+    assert.dom('a#test-list-item').hasAttribute('data-test1');
+    assert.dom('a#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should spread all the attributes passed to the <button> element', async function (assert) {
+    this.set('onClick', () => {});
+
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" @onClick={{this.onClick}} class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('button#test-list-item').hasClass('my-class');
+    assert.dom('button#test-list-item').hasAttribute('data-test1');
+    assert.dom('button#test-list-item').hasAttribute('data-test2', 'test');
+  });
+
+  test('it should yield the content of the List item element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" as |L|><L.Content>test</L.Content></Cut::ListItem>`
+    );
+    assert.dom('div#test-list-item > div.cut-list-item__content').exists();
+    assert
+      .dom('div#test-list-item > div.cut-list-item__content')
+      .hasText('test');
+  });
+
+  test('it should yield the action(button) of the List item element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" as |L|><L.Action as |A|><A.Button @text="test"/></L.Action></Cut::ListItem>`
+    );
+    assert.dom('div#test-list-item > div > button.hds-button').exists();
+    assert.dom('div#test-list-item > div > button.hds-button').hasText('test');
+  });
+
+  test('it should yield the action(generic) of the List item element', async function (assert) {
+    await render(
+      hbs`<Cut::ListItem id="test-list-item" as |L|>
+            <L.Action as |A|>
+              <A.Generic>
+                <pre>test</pre>
+              </A.Generic>
+            </L.Action>
+          </Cut::ListItem>`
+    );
+    assert.dom('div#test-list-item > div > pre').exists();
```

Comment:
as mentioned above: relying too much on the expected DOM structure makes the refactoring quite painful in the future (you're not just testing a feature, that you expect may not change, but also the DOM structure, which instead may be changed without changing the functionality you're testing).